EventListener/DynamicRelationListener.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+/**
+ * This file is part of the FOSCommentBundle package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace FOS\CommentBundle\EventListener;
+
+use Doctrine\Common\Annotations\Reader;
+use Doctrine\Common\EventArgs;
+use Doctrine\Common\EventSubscriber;
+use Doctrine\ORM\Event\LoadClassMetadataEventArgs;
+use Doctrine\ORM\Mapping\ClassMetadataInfo;
+use Symfony\Component\DependencyInjection\Container;
+
+/**
+ * A listener that dynamically adds relations for commentable annotation
+ *
+ * @author Nico Kaag <nico.kaag@genj.nl>
+ */
+class DynamicRelationListener implements EventSubscriber
+{
+    /**
+     * @var Container
+     */
+    protected $container;
+
+    /**
+     * @var AnnotationReader
+     */
+    protected $reader;
+
+    public function __construct(Container $container, Reader $reader) {
```

Comment:
and please don't inject the whole container when you only need to get a single parameter from it. Inject the parameter itself.